src/test/java/com/google/sps/TimeRangeTest.java

```diff
@@ -0,0 +1,85 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.sps.data.*;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashMap;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class TimeRangeTest {
+
+  private static final int DURATION_0_MINUTES = 0;
+  private static final int DURATION_1_HOUR = 60;
+  private static final int DURATION_2_HOUR = 120;
+
+  private static final TimeRange timeRangeA = TimeRange.fromStartDuration(TimeRange.MONDAY, 23, 00, DURATION_2_HOUR);
+  private static final TimeRange timeRangeB = TimeRange.fromStartDuration(TimeRange.TUESDAY, 0, 00, DURATION_2_HOUR);
+  private static final TimeRange timeRangeC = TimeRange.fromStartDuration(TimeRange.WEDNESDAY, 10, 00, DURATION_2_HOUR);
+  private static final TimeRange timeRangeD = TimeRange.fromStartDuration(TimeRange.WEDNESDAY, 12, 00, DURATION_2_HOUR);
+  private static final TimeRange timeRangeE = TimeRange.fromStartDuration(TimeRange.WEDNESDAY, 13, 00, DURATION_2_HOUR);
+  private static final TimeRange timeRangeF = TimeRange.fromStartDuration(TimeRange.SUNDAY, 23, 00, DURATION_2_HOUR);
+  private static final TimeRange timeRangeG = TimeRange.fromStartDuration(TimeRange.MONDAY, 0, 00, DURATION_2_HOUR);
+
+  @Test
+  public void noOverlapSameDayTest() {
+    int overlap = timeRangeC.calculateMinutesOverlap(timeRangeD);
+    int actual = overlap;
+    int expected = DURATION_0_MINUTES;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noOverlapDiffDaysTest() {
+    int overlap = timeRangeB.calculateMinutesOverlap(timeRangeC);
+    int actual = overlap;
+    int expected = DURATION_0_MINUTES;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void normalOverlapTest() {
+    int overlap = timeRangeD.calculateMinutesOverlap(timeRangeE);
+    int actual = overlap;
+    int expected = DURATION_1_HOUR;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void overnightOverlapTest() {
+    int overlap = timeRangeA.calculateMinutesOverlap(timeRangeB);
+    int actual = overlap;
+    int expected = DURATION_1_HOUR;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void sunToMonOverlapTest() {
+    int overlap = timeRangeF.calculateMinutesOverlap(timeRangeG);
+    int actual = overlap;
+    int expected = DURATION_1_HOUR;
+    Assert.assertEquals(expected, actual);
+  }
```

Comment:
Does this test pass? It looks like the implementation might calculate a large negative overlap and return zero here instead of one hour.

Reply:
Yes `sunToMon` works. I realized what the issue was; the week "resets" from Sat -> Sun, so I created tests `satToSun` and it returns exactly what you describe. I added that edgecase check to the function so that it handles it [here](https://github.com/googleinterns/step78-2020/pull/14/commits/40cac5c14f07a96b94b1c0904b1c99c075a84ae6)